homeassistant/components/sensor/radarr.py

```diff
@@ -0,0 +1,246 @@
+    vol.Optional(CONF_INCLUDED, default=[]): cv.ensure_list,
+    vol.Optional(CONF_SSL, default=False): cv.boolean,
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_URLBASE, default=DEFAULT_URLBASE): cv.string,
+    vol.Optional(CONF_DAYS, default=DEFAULT_DAYS): cv.string,
+    vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): vol.In(BYTE_SIZES)
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Radarr platform."""
+    conditions = config.get(CONF_MONITORED_CONDITIONS)
+    add_devices(
+        [RadarrSensor(hass, config, sensor) for sensor in conditions]
+    )
+    return True
+
+
+class RadarrSensor(Entity):
+    """Implemention of the Radarr sensor."""
+
+    def __init__(self, hass, conf, sensor_type):
+        """Create Radarr entity."""
+        from pytz import timezone
+        self.conf = conf
+        self.host = conf.get(CONF_HOST)
+        self.port = conf.get(CONF_PORT)
+        self.urlbase = conf.get(CONF_URLBASE)
+        if self.urlbase:
+            self.urlbase = "%s/" % self.urlbase.strip('/')
+        self.apikey = conf.get(CONF_API_KEY)
+        self.included = conf.get(CONF_INCLUDED)
+        self.days = int(conf.get(CONF_DAYS))
+        self.ssl = 's' if conf.get(CONF_SSL) else ''
+
+        # Object data
+        self.data = []
+        self._tz = timezone(str(hass.config.time_zone))
+        self.type = sensor_type
+        self._name = SENSOR_TYPES[self.type][0]
+        if self.type == 'diskspace':
+            self._unit = conf.get(CONF_UNIT)
+        else:
+            self._unit = SENSOR_TYPES[self.type][1]
+        self._icon = SENSOR_TYPES[self.type][2]
+
+        # Update sensor
+        self._available = False
+        self.update()
+
+    def update(self):
+        """Update the data for the sensor."""
+        start = get_date(self._tz)
+        end = get_date(self._tz, self.days)
+        try:
+            res = requests.get(
+                ENDPOINTS[self.type].format(
+                    self.ssl, self.host, self.port,
+                    self.urlbase, self.apikey, start, end),
+                timeout=5)
+        except OSError:
+            _LOGGER.error('Host %s is not available', self.host)
+            self._available = False
+            self._state = None
+            return
+
+        if res.status_code == 200:
+            if self.type in ['wanted', 'upcoming', 'movies', 'commands']:
+                if self.type == 'movies':
+                    self.data = list(
+                        filter(
+                            lambda x: x['downloaded'],
+                            res.json()
+                        )
+                    )
+                elif self.type == 'wanted':
+                    self.data = list(
+                        filter(
+                            lambda x: not x['downloaded'],
```

Comment:
Wasn't really sure what to do here. Radarr does not have a "wanted" API, so I just filtered movies based on whether they were downloaded or not.